enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -191,7 +201,86 @@ public void unpackResponse( Response<?> response, final TxHandler txHandler ) th
         {
             if ( response.hasTransactionsToBeApplied() )
             {
-                applyQueuedTransactions();
+                applyQueuedTransactionsIfNeeded();
+            }
+        }
+    }
+
+    private void applyQueuedTransactionsIfNeeded() throws IOException
+    {
+        if ( transactionQueue.isEmpty() )
+        {
+            return;
+        }
```

Comment:
Queue can be drained in two cases:
 1) it got full because batch of txs from master was too big
 2) processed response `#hasTransactionsToBeApplied()` returns `true`; this seem to always be true for `TransactionStreamResponse` regardless of transaction stream size

I think (2) is the reason for this check.